contributed_definitions/nyaml/NXmpes.yaml

```diff
@@ -275,6 +275,54 @@ NXmpes(NXobject):
             raw(NX_NUMBER):
               doc: |
                 Raw data before calibration.
+            energy_raw(NX_NUMBER):
```

Comment:
I'm not sure if we want this `..._raw` style of field names. This wouldn't be really energy values but, e.g., tof times or pixel positions for x/kx raw values. So I think on this stage here we would rather insert pixel positions and I think we should also name them such. I think this applies better to a lone NXdetector being written, because providers of a detector might not know whether this was, e.g., an energy or momentum axis.

Reply:
Ok, makes sense. I was also looking at the `NXdetector` base class and there are many fields which are similar to what we want to do, but they don't quite fit the fields that we probably need. 

I am not so familiar with the axes that are typically recorded in ARPES, maybe @rettigl and @Tommaso-Pincelli could comment. How about we start with something like this?

```yaml
(NXdetector):
  (NXdata):
    exists: recommended
    \@signal:
      enumeration: [raw]
    raw(NX_NUMBER):
      doc: |
        Raw data before calibration.
    detector_pixel_x(NX_NUMBER): # there is also x_pixel_offset in NXdetector base class
      doc: |
        Detector pixel in x direction.
      unit: NX_ANY
    detector_pixel_y(NX_NUMBER):
      doc: |
        Detector pixel in y direction.
      unit: NX_ANY
    raw_time_of_flight(NX_INT):
      # exists in NXdetector base class
      # unit: NX_PULSES
      # doc: |
      #   In DAQ clock pulses
    time_of_flight(NX_FLOAT): 
      # exists in NXdetector base class
      # doc: |
      #   Total time of flight 
      # unit: NX_TIME_OF_FLIGHT
    time_of_flight_adc(NX_FLOAT):
      doc: |
        Time-of-flight values, analog-to-digital converted.
      unit: NX_ANY
    polar_angle_manipulator(NX_FLOAT):
      unit: NX_ANGLE
      doc: |
        Polar rotation angle of the manipulator.
    azimuthal_angle_manipulator(NX_FLOAT):
      unit: NX_ANGLE
      doc: |
        Azimuthal rotation angle of the manipulator.
    delay_position(NX_FLOAT):
      doc: |
        Delay position of delay line detector.
      unit: NX_LENGTH
    delay_position_ADC(NX_FLOAT):
      doc: |
        Delay position of delay line detector, analog-to-digital converted.
      unit: NX_ANY
    voltage(NX_FLOAT):
      doc: |
        Measured detector voltage
      unit: NX_VOLTAGE
```
I am not so sure what to do about `magnetization` and `scattering_direction` since these are probably string values and not axes.

Reply:
I think for a DLD we should have at least something for an x-position, y-position and `time_of_flight`. I don't know if there is a difference between `time_of_flight` or `time_of_flight_adc` so people might want to have the option to choose between them here. I think also `delay_position(_ADC)` would refer to the same value as time delay of the signals. We probably want also something like `delay_x` and `delay_y` for the time delay differences of a DLD as they don't really have pixels, but we might also like to allow storing the non-deconvolved delay time relative to the trigger, which would probably give us `time_delay_x` and `time_delay_y`. But @Tommaso-Pincelli and @rettigl probably know much better what is reasonable here.

For a time-integrated detector we probably just want `pixel_x` and `pixel_y` directly from the detector, right?

I was also think w.r.t. not introducing anything out of the scope of NXdetector here to not have `polar_angle_manipulator`, `azimuthal_angle_manipulator` or at least have an additional `external_AXIS`, which describes an axis which is coming from outside the detectors scope (think of a detector just being triggered for readout by the rest of the experimental setup - it would just know that it collected N images, which would flatten the external parameters to one axis, too). This we then could link with NXcalibration to the appropriate fields in the instrument and write it to the top-level NXdata, maybe we should also think of slicing axis in NXcalibration for the case that we need to rearange the dimensions of the external data axis (e.g., it is flat but two parameters where actually varied).

Reply:
Ok, there seem to be mayn open questions, let's wait what the others have to say. For now, I have just removed the raw.. notation and summarized all of the suggestions above in the app def.